### PR TITLE
Fix upload step

### PR DESCRIPTION
**Pull request by KyleFromNVIDIA**
https://github.com/rapidsai/rapids-build-backend/pull/22 changed how the package is uploaded to S3, but did not include `RAPIDS_PY_WHEEL_NAME`. Fix uploading, and also don't publish the `main` branch.

### CODE REVIEW

File: .github/workflows/pr.yaml
Comment (KyleFromNVIDIA) [inline .github/workflows/pr.yaml:17]: This is temporary for testing and will be removed before this is merged.

**Review by vyasr — APPROVED** [review]
Approving assuming you revert the pr.yaml change. We can see that the upload succeeded: https://anaconda.org/rapidsai-wheels-nightly/rapids-build-backend/files.

**Comment by KyleFromNVIDIA** [comment]
I'm wondering if, in this PR, we should also change it so that only tags (and not `main`) get published.

**Comment by KyleFromNVIDIA** [comment]
I've implemented the above.

**Review by jameslamb — APPROVED** [review]
I'm good with only publishing packages on tags, thanks.

**Pull request merged**
